### DISCUSSION TITLE
Possible to set/add log-file "appender"(?) on the fly?

### DISCUSSION THREAD

**Discussion by hoijui** [tokio-rs/tracing#2445] [Q&A]
I tried for quite a while to get this done with a reload layer, similar to https://github.com/tokio-rs/tracing/discussions/2121,
without success.
I though I might be able to add logging to `io::sink` *which implements `Writer`),
and later change that writer to the result of a `File::open("log.txt")?`.
Am also happy about less hacky solutions though.

I just want to initilaize logging to stderr first, and later, potentially, also add logging to a file.

**Comment by davidbarsky** [comment]
I don't think creating a `MakeWriter` than can switch between `stderr` and then writing to a file is hacky _or_ unreasonable—it's why that interface exists.
